### PR TITLE
keep recursing past directories with pubspec.yaml when gathering packages

### PR DESCRIPTION
**Pull request by amirh**

### CODE REVIEW

File: packages/flutter_tools/pubspec.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ dependencies:
   intl: 0.15.2
   json_rpc_2: 2.0.4
   json_schema: 1.0.6
-  linter: 0.1.40
+  linter: 0.1.41
```
Comment (Hixie) [inline packages/flutter_tools/pubspec.yaml]: ?

Comment (Hixie) [reply packages/flutter_tools/pubspec.yaml]: This should surely have affected more than just this package.

**Comment by Hixie** [comment]
Test?

File: packages/flutter_tools/pubspec.yaml (hunk 1 above)
Comment (amirh) [reply packages/flutter_tools/pubspec.yaml]: I ran update-packages, and linter was just updated between the time I submitted the previous update-packages CL and this one

**Comment by Hixie** [comment]
This code is used by more than just update-packages by the way. You may wish to audit the clients and verify that they'll work with the new semantics.

**Comment by amirh** [comment]
Is there any existing test for this method?

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
If there's an existing test, it should fail. If nothing fails, I guess there's no test yet.
(A lot of flutter_tools is undertested right now.)

**Comment by amirh** [comment]
It's used by analyze and update-packages, both seems to make sense with this change

**Comment by amirh** [comment]
Added a test for getRepoPackages

**Pull request merged**